omod/src/test/java/org/openmrs/module/aijar/page/controller/MarkPatientDeadTest.java

```diff
@@ -0,0 +1,103 @@
+        cal.setTime(date);
+        cal.add(Calendar.DATE, daysOff);
+        Date modifiedDated = cal.getTime();
+        return modifiedDated;
+    }
+
+    /**
+     * This tests if the patient is marked as dead when given the right params
+     */
+    @Test
+    public void shouldMarkPatientDeadSuccessfully() {
+        MarkPatientDeadPageController markPatientDeadPageController = new MarkPatientDeadPageController();
+        markPatientDeadPageController.post(concept.getUuid(), true, dateOfDeath, patient.getUuid());
+        Assert.assertEquals(patient.getDead(), true);
+        Assert.assertEquals(patient.getCauseOfDeath(), concept);
+    }
+
+    /**
+     * This tests scenarios where date of death is not given.
+     */
+    @Test
+    public void shouldNotMarkPatientDeadWhenDateIsNull() {
+        MarkPatientDeadPageController markPatientDeadPageController = new MarkPatientDeadPageController();
+        markPatientDeadPageController.post(concept.getUuid(), true, null, patient.getUuid());
+        Assert.assertEquals(patient.getDeathDate(), null);
+    }
+
+
+    /**
+     * This is to test scenarios where a cause of death is not given.
+     */
+    @Test
+    public void shouldNotMarkPatientDeadWhenCauseOfDeathIsNull() {
+        MarkPatientDeadPageController markPatientDeadPageController = new MarkPatientDeadPageController();
+        markPatientDeadPageController.post("", true, dateOfDeath, patient.getUuid());
+        Assert.assertEquals(patient.getDead(), false);
+        Assert.assertNotEquals(patient.getCauseOfDeath(), concept);
+    }
+
+    /**
+     * This tests if the patient is marked as dead when checkbox dead is not checked ie false
+     */
+    @Test
+    public void shouldNotMarkPatientDeadWhenDeadIsFalse() {
+        MarkPatientDeadPageController markPatientDeadPageController = new MarkPatientDeadPageController();
+        markPatientDeadPageController.post(concept.getUuid(), false, dateOfDeath, patient.getUuid());
+        Assert.assertEquals(patient.getDead(), false);
+        Assert.assertNotEquals(patient.getCauseOfDeath(), concept);
+    }
+
+
+    /**
+     * This tests scenarios when date of death is less than birth date
+     */
+    @Test
+    public void deathDateShouldNotBeLessThanBirthDate() {
+        MarkPatientDeadPageController markPatientDeadPageController = new MarkPatientDeadPageController();
+        markPatientDeadPageController.post(concept.getUuid(), true, modifyDate(-30, patient.getBirthdate()), patient.getUuid());
+        Assert.assertNull(patient.getDeathDate());
+    }
+    /**
+     * This tests scenarios when date of death is greater than today
+     */
+    @Test
+    public void deathDateShouldNotBeGreaterThanToday() {
+        MarkPatientDeadPageController markPatientDeadPageController = new MarkPatientDeadPageController();
+        markPatientDeadPageController.post(concept.getUuid(), true, modifyDate(30, dateOfDeath), patient.getUuid());
```

Comment:
@slubwama I had a comment on this before - the dates are hard-coded in the test data, so use actual dates here so that the user later can validate the test cases in case of error, rather than using computations